### ISSUE TITLE
[SmartMeter] input overrun on /dev/ttyUSB0

### ISSUE THREAD

**Issue by morningstar1** [openhab/openhab-addons#11489]
## Expected Behavior
Longterm stable processing of sml messages on a tty.

## Current Behavior
On my raspberry pi 4 I'm using the smartmeter binding sucessfully. But after a least one day it stops working with "Communication Error".
On dmesg I can this this message about every 5 seconds.
```
[  640.539518] ttyUSB0: 5 input overrun(s)
[  640.635623] ttyUSB1: 5 input overrun(s)
```
I'm using two USB Serial converters for my two power meters.
After a restart of openhab, everything works normal again untils it stops at least on day later.

## Possible Solution
I think that some insense processes prevents openhab from processing the serial data. I was not able to increase the kernel buffer size. Therefore I'm using socat to have a software buffer in between.
File `/etc/systemd/system/socat-loopback@.service`
```
[Unit]
Description=Socat Serial Loopback
Binds-to=/dev/tty%i

[Service]
Type=simple
StandardOutput=syslog
StandardError=syslog
SyslogIdentifier=socat-serial-lo

ExecStart=/usr/bin/socat -d -d pty,link=/dev/tty%iSocat,rawer,wait-slave,user=openhab,group=openhab /dev/tty%i,nonblock,rawer 
Restart=always

[Install]
WantedBy=multi-user.target
```
You can start the service with
```
sudo systemctl start socat-loopback@USB0
```
Now you can tell the Smart Meter binding to use `/dev/ttyUSB0Socat` device for reading.
Don't forget to `enable` the service to have it running on startup.

## Your Environment
Raspberry Pi 4 with openhab 3.1 on Raspbian buster

**Comment by wborn** [comment]
> I think that some insense processes prevents openhab from processing the serial data.

Maybe you can also solve this by changing process priorities. That way openHAB will always be able to handle the serial communications which prevents buffers to get full. The other processes then have to wait a bit for their less important tasks. :wink:

**Comment by lsiepel** [comment]
If I’m right, the issue has been solved with  openHAB 3.3.0 

details in this post https://community.openhab.org/t/oh3-x-alternative-java-serial-provider/128462/12

Can you confirm and close this issue?

**Comment by morningstar1** [comment]
I removed my workaround on my raspi and will give openhab 3.4.0 a chance. If I don't see any issues within two weeks, I will happily close this issue.
